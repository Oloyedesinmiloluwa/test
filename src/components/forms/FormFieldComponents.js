import React from 'react';
import PropTypes from 'prop-types';
import { Form, Label } from 'semantic-ui-react';

const FormField = ({
  type, name, onChange, value, placeholder, label, errors,
}) => (
  <Form.Field>
    <label htmlFor={name}>
      {label}
      <input
        type={type}
        id={name}
        name={name}
        onChange={onChange}
        value={value}
        placeholder={placeholder}
      />
    </label>
    {errors && (
      <Label color="red" pointing>
        {errors}
      </Label>
    )}
  </Form.Field>
);

FormField.defaultProps = {
  errors: undefined,
  // onChange: undefined,
};

FormField.propTypes = {
  type: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  errors: PropTypes.string,
};

export default FormField;
