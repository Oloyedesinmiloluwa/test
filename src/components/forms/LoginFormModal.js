import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import {
  Button,
  Dimmer,
  Form,
  Loader,
  Modal,
  TransitionablePortal,
} from 'semantic-ui-react';
import ModalFormHOC, { modalOptions } from '../hoc/ModalFormHOC';
import LoginFields from './FormFieldComponents';

export const LoginForm = ({
  state, show, close, onSubmit, onChange,
}) => {
  const {
    errors,
    loading,
    open,
    user: { email, password },
  } = state;

  return (
    <div>
      <Link href="#" to="#" onClick={show}>
        Login with email
      </Link>
      <TransitionablePortal transition={modalOptions} onClose={close} open={open}>
        <Modal
          centered
          closeIcon
          id="modal-form"
          size="tiny"
          dimmer="blurring"
          open={open}
          onClose={close}
        >
          <Dimmer active={loading}>
            <Loader>Preparing your engagement</Loader>
          </Dimmer>
          <Modal.Header>Welcome</Modal.Header>
          <Modal.Content>
            <Form className="innerForm " size="large" onSubmit={onSubmit} error>
              {/* Rendering the form fields from the FormField component */}
              <LoginFields
                type="email"
                name="email"
                onChange={onChange}
                value={email}
                placeholder="example@email.com"
                label="Email:"
                errors={errors.email}
              />
              <LoginFields
                type="password"
                name="password"
                onChange={onChange}
                value={password}
                placeholder="Enter your password"
                label="Password:"
                erros={errors.password}
              />
              <Button className="btn" type="submit">
                Login
              </Button>
            </Form>
          </Modal.Content>
          <Modal.Actions />
        </Modal>
      </TransitionablePortal>
    </div>
  );
};

LoginForm.propTypes = {
  state: PropTypes.shape({
    user: PropTypes.shape({
      email: PropTypes.string.isRequired,
      password: PropTypes.string.isRequired,
    }).isRequired,
    errors: PropTypes.shape({}).isRequired,
    loading: PropTypes.bool.isRequired,
    open: PropTypes.bool.isRequired,
  }).isRequired,
  show: PropTypes.func.isRequired,
  close: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default ModalFormHOC(LoginForm);
