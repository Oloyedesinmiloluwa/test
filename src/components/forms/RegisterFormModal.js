import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import {
  Button,
  Dimmer,
  Form,
  Loader,
  Modal,
  TransitionablePortal,
} from 'semantic-ui-react';
import ModalFormHOC, { modalOptions } from '../hoc/ModalFormHOC';
import LoginFields from './FormFieldComponents';

export const RegisterForm = ({
  state, show, close, onSubmit, onChange,
}) => {
  const {
    errors,
    loading,
    open,
    user: { email, username, password },
  } = state;

  return (
    <div>
      <Link href="#" to="#" onClick={show}>
        Register
      </Link>
      <TransitionablePortal transition={modalOptions} onClose={close} open={open}>
        <Modal
          centered
          closeIcon
          id="modal-form"
          size="tiny"
          dimmer="blurring"
          open={open}
          onClose={close}
        >
          <Dimmer active={loading}>
            <Loader>Preparing your engagement</Loader>
          </Dimmer>
          <Modal.Header>Welcome register</Modal.Header>
          <Modal.Content>
            <Form className="innerForm " size="large" onSubmit={onSubmit}>
              {/* Rendering the form fields from the FormField component */}
              <LoginFields
                type="text"
                name="username"
                onChange={onChange}
                value={username}
                placeholder="Enter your username"
                label="Username:"
                errors={errors.username}
              />
              <LoginFields
                type="email"
                name="email"
                onChange={onChange}
                value={email}
                placeholder="example@gmail.com"
                label="Email:"
                errors={errors.email}
              />
              <LoginFields
                type="password"
                name="password"
                onChange={onChange}
                value={password}
                placeholder="Enter your password"
                label="Password:"
                errors={errors.password}
              />
              <Button className="btn" type="submit">
                Register
              </Button>
            </Form>
          </Modal.Content>
          <Modal.Actions />
        </Modal>
      </TransitionablePortal>
    </div>
  );
};

RegisterForm.propTypes = {
  state: PropTypes.shape({
    user: PropTypes.shape({
      username: PropTypes.string.isRequired,
      email: PropTypes.string.isRequired,
      password: PropTypes.string.isRequired,
    }).isRequired,
    errors: PropTypes.shape({}).isRequired,
    loading: PropTypes.bool.isRequired,
    open: PropTypes.bool.isRequired,
  }).isRequired,
  show: PropTypes.func.isRequired,
  close: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default ModalFormHOC(RegisterForm);
