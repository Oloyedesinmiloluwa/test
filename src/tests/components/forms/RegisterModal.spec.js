import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import { RegisterForm } from '../../../components/forms/RegisterFormModal';

describe(' LoginModal component rendering', () => {
  const state = {
    user: {
      username: '',
      email: '',
      password: '',
    },
    errors: {
      username: 'Invalid',
      email: 'Invalid',
      password: 'Invalid',
    },
    loading: false,
    open: false,
  };
  const sinonSpy = sinon.spy();
  const wrapper = shallow(<RegisterForm
    state={state}
    show={sinonSpy}
    close={sinonSpy}
    onSubmit={sinonSpy}
    onChange={sinonSpy}
  />);

  it('Should render properly', () => {
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should call the submit method on submit', () => {
    wrapper.find('Form').simulate('submit', { preventDefault() {} });
    expect(sinonSpy.called).toBe(true);
  });
});
