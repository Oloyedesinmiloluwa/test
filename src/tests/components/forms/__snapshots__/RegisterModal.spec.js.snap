// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` LoginModal component rendering Should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RegisterForm
    close={[Function]}
    onChange={[Function]}
    onSubmit={[Function]}
    show={[Function]}
    state={
      Object {
        "errors": Object {
          "email": "Invalid",
          "password": "Invalid",
          "username": "Invalid",
        },
        "loading": false,
        "open": false,
        "user": Object {
          "email": "",
          "password": "",
          "username": "",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          href="#"
          onClick={[Function]}
          replace={false}
          to="#"
        >
          Register
        </Link>,
        <TransitionablePortal
          onClose={[Function]}
          open={false}
          transition={
            Object {
              "animation": "fly up",
              "duration": 1000,
            }
          }
        >
          <Modal
            centered={true}
            closeIcon={true}
            closeOnDimmerClick={true}
            closeOnDocumentClick={false}
            dimmer="blurring"
            eventPool="Modal"
            id="modal-form"
            onClose={[Function]}
            open={false}
            size="tiny"
          >
            <Dimmer
              active={false}
            >
              <Loader>
                Preparing your engagement
              </Loader>
            </Dimmer>
            <ModalHeader>
              Welcome register
            </ModalHeader>
            <ModalContent>
              <Form
                as="form"
                className="innerForm "
                onSubmit={[Function]}
                size="large"
              >
                <FormField
                  errors="Invalid"
                  label="Username:"
                  name="username"
                  onChange={[Function]}
                  placeholder="Enter your username"
                  type="text"
                  value=""
                />
                <FormField
                  errors="Invalid"
                  label="Email:"
                  name="email"
                  onChange={[Function]}
                  placeholder="example@gmail.com"
                  type="email"
                  value=""
                />
                <FormField
                  errors="Invalid"
                  label="Password:"
                  name="password"
                  onChange={[Function]}
                  placeholder="Enter your password"
                  type="password"
                  value=""
                />
                <Button
                  as="button"
                  className="btn"
                  role="button"
                  type="submit"
                >
                  Register
                </Button>
              </Form>
            </ModalContent>
            <ModalActions />
          </Modal>
        </TransitionablePortal>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Register",
          "href": "#",
          "onClick": [Function],
          "replace": false,
          "to": "#",
        },
        "ref": null,
        "rendered": "Register",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Modal
            centered={true}
            closeIcon={true}
            closeOnDimmerClick={true}
            closeOnDocumentClick={false}
            dimmer="blurring"
            eventPool="Modal"
            id="modal-form"
            onClose={[Function]}
            open={false}
            size="tiny"
          >
            <Dimmer
              active={false}
            >
              <Loader>
                Preparing your engagement
              </Loader>
            </Dimmer>
            <ModalHeader>
              Welcome register
            </ModalHeader>
            <ModalContent>
              <Form
                as="form"
                className="innerForm "
                onSubmit={[Function]}
                size="large"
              >
                <FormField
                  errors="Invalid"
                  label="Username:"
                  name="username"
                  onChange={[Function]}
                  placeholder="Enter your username"
                  type="text"
                  value=""
                />
                <FormField
                  errors="Invalid"
                  label="Email:"
                  name="email"
                  onChange={[Function]}
                  placeholder="example@gmail.com"
                  type="email"
                  value=""
                />
                <FormField
                  errors="Invalid"
                  label="Password:"
                  name="password"
                  onChange={[Function]}
                  placeholder="Enter your password"
                  type="password"
                  value=""
                />
                <Button
                  as="button"
                  className="btn"
                  role="button"
                  type="submit"
                >
                  Register
                </Button>
              </Form>
            </ModalContent>
            <ModalActions />
          </Modal>,
          "onClose": [Function],
          "open": false,
          "transition": Object {
            "animation": "fly up",
            "duration": 1000,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "centered": true,
            "children": Array [
              <Dimmer
                active={false}
              >
                <Loader>
                  Preparing your engagement
                </Loader>
              </Dimmer>,
              <ModalHeader>
                Welcome register
              </ModalHeader>,
              <ModalContent>
                <Form
                  as="form"
                  className="innerForm "
                  onSubmit={[Function]}
                  size="large"
                >
                  <FormField
                    errors="Invalid"
                    label="Username:"
                    name="username"
                    onChange={[Function]}
                    placeholder="Enter your username"
                    type="text"
                    value=""
                  />
                  <FormField
                    errors="Invalid"
                    label="Email:"
                    name="email"
                    onChange={[Function]}
                    placeholder="example@gmail.com"
                    type="email"
                    value=""
                  />
                  <FormField
                    errors="Invalid"
                    label="Password:"
                    name="password"
                    onChange={[Function]}
                    placeholder="Enter your password"
                    type="password"
                    value=""
                  />
                  <Button
                    as="button"
                    className="btn"
                    role="button"
                    type="submit"
                  >
                    Register
                  </Button>
                </Form>
              </ModalContent>,
              <ModalActions />,
            ],
            "closeIcon": true,
            "closeOnDimmerClick": true,
            "closeOnDocumentClick": false,
            "dimmer": "blurring",
            "eventPool": "Modal",
            "id": "modal-form",
            "onClose": [Function],
            "open": false,
            "size": "tiny",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "children": <Loader>
                  Preparing your engagement
                </Loader>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Preparing your engagement",
                },
                "ref": null,
                "rendered": "Preparing your engagement",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Welcome register",
              },
              "ref": null,
              "rendered": "Welcome register",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Form
                  as="form"
                  className="innerForm "
                  onSubmit={[Function]}
                  size="large"
                >
                  <FormField
                    errors="Invalid"
                    label="Username:"
                    name="username"
                    onChange={[Function]}
                    placeholder="Enter your username"
                    type="text"
                    value=""
                  />
                  <FormField
                    errors="Invalid"
                    label="Email:"
                    name="email"
                    onChange={[Function]}
                    placeholder="example@gmail.com"
                    type="email"
                    value=""
                  />
                  <FormField
                    errors="Invalid"
                    label="Password:"
                    name="password"
                    onChange={[Function]}
                    placeholder="Enter your password"
                    type="password"
                    value=""
                  />
                  <Button
                    as="button"
                    className="btn"
                    role="button"
                    type="submit"
                  >
                    Register
                  </Button>
                </Form>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "as": "form",
                  "children": Array [
                    <FormField
                      errors="Invalid"
                      label="Username:"
                      name="username"
                      onChange={[Function]}
                      placeholder="Enter your username"
                      type="text"
                      value=""
                    />,
                    <FormField
                      errors="Invalid"
                      label="Email:"
                      name="email"
                      onChange={[Function]}
                      placeholder="example@gmail.com"
                      type="email"
                      value=""
                    />,
                    <FormField
                      errors="Invalid"
                      label="Password:"
                      name="password"
                      onChange={[Function]}
                      placeholder="Enter your password"
                      type="password"
                      value=""
                    />,
                    <Button
                      as="button"
                      className="btn"
                      role="button"
                      type="submit"
                    >
                      Register
                    </Button>,
                  ],
                  "className": "innerForm ",
                  "onSubmit": [Function],
                  "size": "large",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "errors": "Invalid",
                      "label": "Username:",
                      "name": "username",
                      "onChange": [Function],
                      "placeholder": "Enter your username",
                      "type": "text",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "errors": "Invalid",
                      "label": "Email:",
                      "name": "email",
                      "onChange": [Function],
                      "placeholder": "example@gmail.com",
                      "type": "email",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "errors": "Invalid",
                      "label": "Password:",
                      "name": "password",
                      "onChange": [Function],
                      "placeholder": "Enter your password",
                      "type": "password",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "as": "button",
                      "children": "Register",
                      "className": "btn",
                      "role": "button",
                      "type": "submit",
                    },
                    "ref": null,
                    "rendered": "Register",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            href="#"
            onClick={[Function]}
            replace={false}
            to="#"
          >
            Register
          </Link>,
          <TransitionablePortal
            onClose={[Function]}
            open={false}
            transition={
              Object {
                "animation": "fly up",
                "duration": 1000,
              }
            }
          >
            <Modal
              centered={true}
              closeIcon={true}
              closeOnDimmerClick={true}
              closeOnDocumentClick={false}
              dimmer="blurring"
              eventPool="Modal"
              id="modal-form"
              onClose={[Function]}
              open={false}
              size="tiny"
            >
              <Dimmer
                active={false}
              >
                <Loader>
                  Preparing your engagement
                </Loader>
              </Dimmer>
              <ModalHeader>
                Welcome register
              </ModalHeader>
              <ModalContent>
                <Form
                  as="form"
                  className="innerForm "
                  onSubmit={[Function]}
                  size="large"
                >
                  <FormField
                    errors="Invalid"
                    label="Username:"
                    name="username"
                    onChange={[Function]}
                    placeholder="Enter your username"
                    type="text"
                    value=""
                  />
                  <FormField
                    errors="Invalid"
                    label="Email:"
                    name="email"
                    onChange={[Function]}
                    placeholder="example@gmail.com"
                    type="email"
                    value=""
                  />
                  <FormField
                    errors="Invalid"
                    label="Password:"
                    name="password"
                    onChange={[Function]}
                    placeholder="Enter your password"
                    type="password"
                    value=""
                  />
                  <Button
                    as="button"
                    className="btn"
                    role="button"
                    type="submit"
                  >
                    Register
                  </Button>
                </Form>
              </ModalContent>
              <ModalActions />
            </Modal>
          </TransitionablePortal>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Register",
            "href": "#",
            "onClick": [Function],
            "replace": false,
            "to": "#",
          },
          "ref": null,
          "rendered": "Register",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Modal
              centered={true}
              closeIcon={true}
              closeOnDimmerClick={true}
              closeOnDocumentClick={false}
              dimmer="blurring"
              eventPool="Modal"
              id="modal-form"
              onClose={[Function]}
              open={false}
              size="tiny"
            >
              <Dimmer
                active={false}
              >
                <Loader>
                  Preparing your engagement
                </Loader>
              </Dimmer>
              <ModalHeader>
                Welcome register
              </ModalHeader>
              <ModalContent>
                <Form
                  as="form"
                  className="innerForm "
                  onSubmit={[Function]}
                  size="large"
                >
                  <FormField
                    errors="Invalid"
                    label="Username:"
                    name="username"
                    onChange={[Function]}
                    placeholder="Enter your username"
                    type="text"
                    value=""
                  />
                  <FormField
                    errors="Invalid"
                    label="Email:"
                    name="email"
                    onChange={[Function]}
                    placeholder="example@gmail.com"
                    type="email"
                    value=""
                  />
                  <FormField
                    errors="Invalid"
                    label="Password:"
                    name="password"
                    onChange={[Function]}
                    placeholder="Enter your password"
                    type="password"
                    value=""
                  />
                  <Button
                    as="button"
                    className="btn"
                    role="button"
                    type="submit"
                  >
                    Register
                  </Button>
                </Form>
              </ModalContent>
              <ModalActions />
            </Modal>,
            "onClose": [Function],
            "open": false,
            "transition": Object {
              "animation": "fly up",
              "duration": 1000,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "centered": true,
              "children": Array [
                <Dimmer
                  active={false}
                >
                  <Loader>
                    Preparing your engagement
                  </Loader>
                </Dimmer>,
                <ModalHeader>
                  Welcome register
                </ModalHeader>,
                <ModalContent>
                  <Form
                    as="form"
                    className="innerForm "
                    onSubmit={[Function]}
                    size="large"
                  >
                    <FormField
                      errors="Invalid"
                      label="Username:"
                      name="username"
                      onChange={[Function]}
                      placeholder="Enter your username"
                      type="text"
                      value=""
                    />
                    <FormField
                      errors="Invalid"
                      label="Email:"
                      name="email"
                      onChange={[Function]}
                      placeholder="example@gmail.com"
                      type="email"
                      value=""
                    />
                    <FormField
                      errors="Invalid"
                      label="Password:"
                      name="password"
                      onChange={[Function]}
                      placeholder="Enter your password"
                      type="password"
                      value=""
                    />
                    <Button
                      as="button"
                      className="btn"
                      role="button"
                      type="submit"
                    >
                      Register
                    </Button>
                  </Form>
                </ModalContent>,
                <ModalActions />,
              ],
              "closeIcon": true,
              "closeOnDimmerClick": true,
              "closeOnDocumentClick": false,
              "dimmer": "blurring",
              "eventPool": "Modal",
              "id": "modal-form",
              "onClose": [Function],
              "open": false,
              "size": "tiny",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "children": <Loader>
                    Preparing your engagement
                  </Loader>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "Preparing your engagement",
                  },
                  "ref": null,
                  "rendered": "Preparing your engagement",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Welcome register",
                },
                "ref": null,
                "rendered": "Welcome register",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <Form
                    as="form"
                    className="innerForm "
                    onSubmit={[Function]}
                    size="large"
                  >
                    <FormField
                      errors="Invalid"
                      label="Username:"
                      name="username"
                      onChange={[Function]}
                      placeholder="Enter your username"
                      type="text"
                      value=""
                    />
                    <FormField
                      errors="Invalid"
                      label="Email:"
                      name="email"
                      onChange={[Function]}
                      placeholder="example@gmail.com"
                      type="email"
                      value=""
                    />
                    <FormField
                      errors="Invalid"
                      label="Password:"
                      name="password"
                      onChange={[Function]}
                      placeholder="Enter your password"
                      type="password"
                      value=""
                    />
                    <Button
                      as="button"
                      className="btn"
                      role="button"
                      type="submit"
                    >
                      Register
                    </Button>
                  </Form>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "as": "form",
                    "children": Array [
                      <FormField
                        errors="Invalid"
                        label="Username:"
                        name="username"
                        onChange={[Function]}
                        placeholder="Enter your username"
                        type="text"
                        value=""
                      />,
                      <FormField
                        errors="Invalid"
                        label="Email:"
                        name="email"
                        onChange={[Function]}
                        placeholder="example@gmail.com"
                        type="email"
                        value=""
                      />,
                      <FormField
                        errors="Invalid"
                        label="Password:"
                        name="password"
                        onChange={[Function]}
                        placeholder="Enter your password"
                        type="password"
                        value=""
                      />,
                      <Button
                        as="button"
                        className="btn"
                        role="button"
                        type="submit"
                      >
                        Register
                      </Button>,
                    ],
                    "className": "innerForm ",
                    "onSubmit": [Function],
                    "size": "large",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "errors": "Invalid",
                        "label": "Username:",
                        "name": "username",
                        "onChange": [Function],
                        "placeholder": "Enter your username",
                        "type": "text",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "errors": "Invalid",
                        "label": "Email:",
                        "name": "email",
                        "onChange": [Function],
                        "placeholder": "example@gmail.com",
                        "type": "email",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "errors": "Invalid",
                        "label": "Password:",
                        "name": "password",
                        "onChange": [Function],
                        "placeholder": "Enter your password",
                        "type": "password",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "as": "button",
                        "children": "Register",
                        "className": "btn",
                        "role": "button",
                        "type": "submit",
                      },
                      "ref": null,
                      "rendered": "Register",
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
